backend/src/main/java/com/bakdata/conquery/io/result/arrow/ArrowRenderer.java

```diff
@@ -0,0 +1,285 @@
+package com.bakdata.conquery.io.result.arrow;
+
+import static com.bakdata.conquery.io.result.arrow.ArrowUtil.NAMED_FIELD_DATE_DAY;
+import static com.bakdata.conquery.io.result.arrow.ArrowUtil.ROOT_ALLOCATOR;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.bakdata.conquery.models.execution.ManagedExecution;
+import com.bakdata.conquery.models.forms.managed.ManagedForm;
+import com.bakdata.conquery.models.query.ManagedQuery;
+import com.bakdata.conquery.models.query.PrintSettings;
+import com.bakdata.conquery.models.query.resultinfo.ResultInfo;
+import com.bakdata.conquery.models.query.results.ContainedEntityResult;
+import com.bakdata.conquery.models.query.results.EntityResult;
+import com.google.common.collect.ImmutableList;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.arrow.util.Preconditions;
+import org.apache.arrow.vector.BitVector;
+import org.apache.arrow.vector.DateDayVector;
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.Float4Vector;
+import org.apache.arrow.vector.Float8Vector;
+import org.apache.arrow.vector.IntVector;
+import org.apache.arrow.vector.VarCharVector;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.arrow.vector.ipc.ArrowWriter;
+import org.apache.arrow.vector.types.FloatingPointPrecision;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.FieldType;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.apache.arrow.vector.util.Text;
+
+@Slf4j
+public class ArrowRenderer {
+	
+	public static void renderToStream(
+		Function<VectorSchemaRoot, ArrowWriter> writerProducer, 
+		PrintSettings cfg,
+		ManagedExecution<?> exec, 
+		Function<ContainedEntityResult,String[]> idMapper, 
+		String[] idHeaders,	
+		int batchsize) throws IOException
+	{
+		// Test the execution if the result is renderable into one table
+		Stream<EntityResult> results;
+		List<ResultInfo> resultInfos;
+		if(exec instanceof ManagedQuery) {
```

Comment:
in Methode kapseln, das collectResultInfos müsstest du doch eigentlich nicht so umständlich benutzen?

Reply:
okay ich mache split phase. Aber ja die Result infos sind nur auf der ManagedQuery definiert